From 88672dbc303b68bb3ad3edfec3446473b4daacbf Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Tue, 23 Mar 2010 15:27:48 +0100
Subject: [PATCH 2/2] Fix keyboard indicator displaying

Always call show_hide_icon() in apply_xkb_settings(), not just when we set a
different keyboard layout. We always want to show the indicator if we have
multiple keyboard layouts available.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=613666
---
 plugins/keyboard/gsd-keyboard-xkb.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/plugins/keyboard/gsd-keyboard-xkb.c b/plugins/keyboard/gsd-keyboard-xkb.c
index bbed91d..753af39 100644
--- a/plugins/keyboard/gsd-keyboard-xkb.c
+++ b/plugins/keyboard/gsd-keyboard-xkb.c
@@ -472,7 +472,6 @@ try_activating_xkb_config_if_new (GkbdKeyboardConfig *
 	if (!gkbd_keyboard_config_equals
 	    (&current_kbd_config, current_sys_kbd_config)) {
 		if (gkbd_keyboard_config_activate (&current_kbd_config)) {
-			show_hide_icon ();
 			if (pa_callback != NULL) {
 				(*pa_callback) (pa_callback_user_data);
 				return TRUE;
@@ -703,6 +702,7 @@ apply_xkb_settings (void)
 		xkl_engine_lock_group (current_config.engine,
 				       group_to_activate);
 	gkbd_keyboard_config_term (&current_sys_kbd_config);
+	show_hide_icon ();
 }
 
 static void
-- 
1.7.0

